hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionConfiguration.java

```diff
@@ -116,6 +142,20 @@
 
     this.writeRpcTimeout = conf.getInt(HConstants.HBASE_RPC_WRITE_TIMEOUT_KEY,
         conf.getInt(HConstants.HBASE_RPC_TIMEOUT_KEY, HConstants.DEFAULT_HBASE_RPC_TIMEOUT));
+
+    long pauseMs = conf.getLong(HBASE_CLIENT_PAUSE, DEFAULT_HBASE_CLIENT_PAUSE);
+    long pauseMsForServerOverloaded = conf.getLong(HBASE_CLIENT_PAUSE_FOR_SERVER_OVERLOADED,
+      conf.getLong(HConstants.HBASE_CLIENT_PAUSE_FOR_CQTBE, pauseMs));
+    if (pauseMsForServerOverloaded < pauseMs) {
```

Comment:
Okay this is strange. This logic (which I think is good) is replicated from the async client. However, on master, the sync client is implemented on top of the async client, so I think this will at best produce the warning message twice, and otherwise have no effect. Or maybe there are places where the ConnectionConfiguration is used, even with the sync-over-async implementation?

It may be that this is someplace where your patches for master and branch-2[,.*] will differ, because on the others, async and sync client implementations are completely independent.

Reply:
Ok cool, i wasn't sure what the best way to handle this is. Honestly, i didn't think about the double-warn, so good catch. What I'll do is remove this from ConnectionConfiguration here in master. So branch-2x will have this slight divergence where it needs to have the logic in both classes, in addition to handling all of the classes in branch-2x that don't exist here anymore.